js/models/messages.js

```diff
@@ -1935,105 +1977,212 @@
-
-            if (!fromAdmin) {
-              // Make sure the message is not removing members / renaming the group
-              const nameChanged =
-                conversation.get('name') !== initialMessage.group.name;
-
-              if (nameChanged) {
-                window.log.warn(
-                  'Non-admin attempts to change the name of the group'
-                );
-              }
+      if (
+        initialMessage.group.members &&
+        initialMessage.group.type === GROUP_TYPES.UPDATE
+      ) {
+        if (newGroup) {
+          conversation.updateGroupAdmins(initialMessage.group.admins);
+
+          conversation.setFriendRequestStatus(
+            window.friends.friendRequestStatusEnum.friends
+          );
+        } else {
+          // be sure to drop a message from a non admin if it tries to change group members
+          // or change the group name
+          const fromAdmin = conversation
+            .get('groupAdmins')
+            .includes(primarySource);
+
+          if (!fromAdmin) {
+            // Make sure the message is not removing members / renaming the group
+            const nameChanged =
+              conversation.get('name') !== initialMessage.group.name;
+
+            if (nameChanged) {
+              window.log.warn(
+                'Non-admin attempts to change the name of the group'
+              );
+            }
 
-              const membersMissing =
-                _.difference(
-                  conversation.get('members'),
-                  initialMessage.group.members
-                ).length > 0;
+            const membersMissing =
+              _.difference(
+                conversation.get('members'),
+                initialMessage.group.members
+              ).length > 0;
 
-              if (membersMissing) {
-                window.log.warn('Non-admin attempts to remove group members');
-              }
+            if (membersMissing) {
+              window.log.warn('Non-admin attempts to remove group members');
+            }
 
-              const messageAllowed = !nameChanged && !membersMissing;
+            const messageAllowed = !nameChanged && !membersMissing;
 
-              if (!messageAllowed) {
-                confirm();
-                return null;
-              }
+            // Returning true drops the message
+            if (!messageAllowed) {
+              confirm();
+              return true;
             }
           }
-          // For every member, see if we need to establish a session:
-          initialMessage.group.members.forEach(memberPubKey => {
-            const haveSession = _.some(
-              textsecure.storage.protocol.sessions,
-              s => s.number === memberPubKey
-            );
+        }
+        // send a session request for all the members we do not have a session with
+        window.libloki.api.sendSessionRequestsToMembers(
+          initialMessage.group.members
+        );
```

Comment:
on group update (not sync events).  trigger a session request with existing members. (except the one we already have one of course)